vdb-bench-assembly/plugins/vdb-bench-dataservice/datasources/datasourceSummaryController.js

```diff
@@ -340,13 +340,34 @@
          * Handle edit ServiceSource click
          */
         var editSvcSourceClicked = function( ) {
-            // Start refresh of Connections and Translators, then broadcast a page change
-            SvcSourceSelectionService.refreshConnectionsAndTranslators();
-            //
-            // Broadcast the pageChange
-            $rootScope.$broadcast("dataServicePageChanged", 'svcsource-edit');
-        };
+            // Get the connectionName and TranslatorName for the selected source,
+            // then transfer to the edit page.
+        	
+            var successCallback = function(model) {
+                // Update the connection name for the source being edited
+                SvcSourceSelectionService.setEditSourceConnectionNameSelection(model.keng__id);
+                
+                var modelSourceSuccessCallback = function(modelSource) {
+                    TranslatorSelectionService.selectTranslatorName(modelSource.vdb__sourceTranslator);
+                    SvcSourceSelectionService.setEditSourceConnectionJndiSelection(modelSource.vdb__sourceJndiName);
+
+                    //
+                    // Broadcast the pageChange
+                    $rootScope.$broadcast("dataServicePageChanged", 'svcsource-edit');
+                };
+                var modelSourceFailureCallback = function(modelSourceErrorMsg) {
+                    alert("Failed to get model source: \n"+modelSourceErrorMsg);
```

Comment:
Thinking that we might need to uniform method of displaying errors. Sometimes, I have used console.log(), other times alert in the same way here. Perhaps a central logger & modal display service?